src/bin/rover.rs

```diff
@@ -1,12 +1,36 @@
 use anyhow::Result;
 use rover::*;
+use sputnik::Session;
 use structopt::StructOpt;
 
+use std::thread;
+
 fn main() -> Result<()> {
     logger::init();
+    let app = cli::Rover::from_args();
+
+    // attempt to create a new `Session` to capture anonymous usage data
+    match Session::new(&app) {
+        // if successful, report the usage data in the background
+        Ok(session) => {
+            // kicks off the reporting on a background thread
+            let report_thread = thread::spawn(move || {
+                // ignore errors because it is not in the critical path
+                let _ = session.report();
+            });
+
+            // kicks off the app on the main thread
+            app.run()?;
```

Comment:
How would you add custom session information from the invocation of a command (like "this introspection request was for a federated graph") with this setup?

Reply:
gonna mark this one as out of scope/needs further discussion for now. we can revisit later. for now it seems to me like a separate concern and would have a different reporting mechanism entirely as it's quite different from the anonymous usage data we're currently collecting.